examples/todo/js/components/Todo.js

```diff
@@ -10,30 +10,41 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+import classnames from 'classnames';
+import React from 'react';
+import Relay from 'react-relay';
+import * as RelaySubscriptions from 'relay-subscriptions';
+
 import ChangeTodoStatusMutation from '../mutations/ChangeTodoStatusMutation';
 import RemoveTodoMutation from '../mutations/RemoveTodoMutation';
 import RenameTodoMutation from '../mutations/RenameTodoMutation';
-import TodoTextInput from './TodoTextInput';
-import * as RelaySubscriptions from 'relay-subscriptions';
 import UpdateTodoSubscription from '../subscriptions/UpdateTodoSubscription';
-
-import React from 'react';
-import Relay from 'react-relay';
-import classnames from 'classnames';
+import TodoTextInput from './TodoTextInput';
 
 class Todo extends React.Component {
+  static propTypes = {
+    viewer: React.PropTypes.object.isRequired,
+    todo: React.PropTypes.object.isRequired,
+    relay: React.PropTypes.object.isRequired,
+    subscriptions: React.PropTypes.object.isRequired,
+  };
+
   state = {
     isEditing: false,
   };
+
   componentDidMount() {
-    this._updateSubscription = this.props.subscriptions.subscribe(
-      new UpdateTodoSubscription({ todo: this.props.todo }),
-      `update_todo_${this.props.todo.id}`
-    );
+    if (!this.props.relay.hasOptimisticUpdate(this.props.todo)) {
```

Comment:
Well, `hasOptimisticUpdate` will be true even when the optimistic update isn't for a newly added node, but that shouldn't ever be the case when the `<Todo>` is mounting. I guess it'd be more correct to check this on the edge on the parent.